ppinot4py/computers/metrics_computer.py

```diff
@@ -131,35 +141,74 @@ def _first_to_after_from(from_condition, to_condition, id_values, ts_values):
     # Returns the first of them
     return df.groupby('id')['t'].first()
 
-def _linear_calculation(fromValues, toValues):
+def _linear_calculation(fromValues, toValues, measure):
     
-    finalResult = toValues - fromValues   
-    finalResult[finalResult < pd.Timedelta(0)] = pd.Timedelta('nan')
+    # Hour, min and second of start and close hour
+
+    if(measure.business_duration != None):
+        open_time= measure.business_duration.business_start
+        close_time= measure.business_duration.business_end
+        
+        # List of holidays
+        holiday_list = measure.business_duration.holiday_list
+        unit_hour= measure.business_duration.unit_hour
+        weekend_list = measure.business_duration.weekend_list
+
+        fromValues.name = 'start'
+        toValues.name = 'end'
 
+        dataf = pd.concat([fromValues, toValues], axis=1)
+        dataf['start'] = pd.to_datetime(dataf['start'], format='%Y-%m-%d %H:%M:%S')
```

Comment:
are you sure that you need these conversion? I think at this point fromValues and toValues should be datetime already.